### PR TITLE
Renaming to Sling Model Persist, fixed test error with JDK 11

### PR DESCRIPTION
**Pull request by badvision**
PostConstruct wasn't imported, needed to add module dependency on javax annotations

### CODE REVIEW

**Comment by badvision** [comment]
@klcodanr -- Renamed by popular request to something that didn't have "JCR" in the name. :)

**Comment by badvision** [comment]
@cziegeler please advise if any more changed are needed.

File: SlingModelPersist/pom.xml [hunk 1]
```diff
@@ -1,176 +1,183 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.apache.sling</groupId>
+    <artifactId>org.apache.sling.models.persist</artifactId>
+    <version>1.0-SNAPSHOT</version>
+    <packaging>bundle</packaging>
+
+    <name>Sling Model Persist OSGi Bundle</name>
```
Comment (klcodanr) [inline SlingModelPersist/pom.xml]: Remove "OSGi Bundle"

File: SlingModelPersist/src/main/java/org/apache/sling/models/persist/impl/util/AssertUtils.java [hunk 2]
```diff
@@ -1,50 +1,50 @@
...
+package org.apache.sling.models.persist.impl.util;
+
+/**
+ * Simple assertion functions.
+ *
+ * Part of the code of this class has been borrowed from the open-source project
+ * <code>jerry-core</code> from https://github.com/sangupta/jerry-core.
+ *
+ */
+public class AssertUtils {
```
Comment (klcodanr) [inline SlingModelPersist/src/main/java/org/apache/sling/models/persist/impl/util/AssertUtils.java]: Is there any reason to not use Commons Lang3

File: SlingModelPersist/src/main/java/org/apache/sling/models/persist/ModelPersist.java [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
+
+import javax.jcr.RepositoryException;
+import org.apache.sling.api.resource.PersistenceException;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Definition of JCR Persist service
+ */
+public interface ModelPersist {
```
Comment (klcodanr) [inline SlingModelPersist/src/main/java/org/apache/sling/models/persist/ModelPersist.java]: For some reason the name ModelPersist seems... off? PersistModel, ModelPersistence?

File: SlingModelPersist/src/test/java/org/apache/sling/models/persist/bean/BeanWithPathField.java [hunk 4]
```diff
@@ -1,23 +1,23 @@
...
-    @Inject
-    public String prop1 = "testValue";
-    
-    public String path = "/test/field-path";
-}
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
```
Comment (klcodanr) [inline SlingModelPersist/src/test/java/org/apache/sling/models/persist/bean/BeanWithPathField.java]: Don't make Apache Rat mad ;-)

File: SlingModelPersist/src/main/java/org/apache/sling/models/persist/impl/util/AssertUtils.java (hunk 2 above)
Comment (badvision) [reply SlingModelPersist/src/main/java/org/apache/sling/models/persist/impl/util/AssertUtils.java]: No reason, I just didn't know if there was a significant difference. :)

File: SlingModelPersist/src/main/java/org/apache/sling/models/persist/ModelPersist.java (hunk 3 above)
Comment (badvision) [reply SlingModelPersist/src/main/java/org/apache/sling/models/persist/ModelPersist.java]: Well I think you might be right.  Class names work better as nouns, and this is more verb-y.  ModelPersistor is something which persists models, maybe that's a better name?

File: SlingModelPersist/src/test/java/org/apache/sling/models/persist/bean/BeanWithPathField.java (hunk 4 above)
Comment (badvision) [reply SlingModelPersist/src/test/java/org/apache/sling/models/persist/bean/BeanWithPathField.java]: GAH!  Good eye.  Sorry.

File: SlingModelPersist/pom.xml [hunk 5]
```diff
@@ -1,176 +1,183 @@
...
-                </configuration>
-            </plugin>
-        </plugins>
-    </build>
-</project>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.apache.sling</groupId>
```
Comment (cziegeler) [inline SlingModelPersist/pom.xml]: You should use Sling's (bundle) parent pom. This way you can remove some of the stuff from your pom

**Pull request merged**